apis/src/main/java/xyz/bookself/controllers/user/BookListController.java

```diff
@@ -57,41 +57,35 @@ public BookListController(BookselfApiConfiguration configuration, BookListReposi
         return new ResponseEntity<>(booksInList, HttpStatus.OK);
     }
 
-    /**
-     * Update a book list (shelf).
-     *  - Rename shelf
-     *  - Add books
-     *  - Remove books
-     * Check if fields are not null before updating shelf.
-     * Unrecognized fields are ignored.
-     * @param shelfDto JSON of the format:
-     *      {
-     *          "newListName": "New Name",
-     *          "booksToBeAdded": [ "book-id-1", "book-id-2" ],
-     *          "booksToBeRemoved": [ "book-id-3", "book-id-4" ]
-     *      }
-     * @param id the shelf id
-     * The correct request method for updating is PUT
-     */
-    @PutMapping(value = "/{id}/update", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<BookList> renameShelf(@PathVariable String id, @RequestBody ShelfDto shelfDto) {
-
-        final String newListName = shelfDto.getNewListName();
-        final Set<String> booksToBeAdded = shelfDto.getBooksToBeAdded();
-        final Set<String> booksToBeRemoved = shelfDto.getBooksToBeRemoved();
-        final String newBookListId = shelfDto.getNewBookListId();
+    @PutMapping(value = "/{id}/update-list-name", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<BookList> renameShelf(@PathVariable String id, @RequestBody ShelfNameDTO shelfNameDTO) {
 
+        final String newListName = shelfNameDTO.getNewListName();
 
         final BookList shelf = bookListRepository.findById(id).orElseThrow();
-        final BookList addShelf = bookListRepository.findById(newBookListId).orElseThrow();
         boolean updated = false;
 
         if(Objects.nonNull(newListName)) {
             shelf.setBookListName(newListName);
             updated = true;
         }
 
-        if(Objects.nonNull(booksToBeAdded) && !booksToBeAdded.isEmpty()) {
+        return new ResponseEntity<>((updated) ? bookListRepository.save(shelf) : shelf, HttpStatus.OK);
+    }
+
+    @PutMapping(value = "/{id}/move-books", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<BookList> updateShelf(@PathVariable String id, @RequestBody MoveShelfDTO moveShelfDto) {
```

Comment:
Hmm consider combining this with the PUT to `/update-list-name` endpoint above. That way you could just have a PUT or PATCH to `/{id}`.

Then you can set up the DTO to just update as needed based on what values are sent in. I.e. if name is sent in without the two sets then just update name. If all three are sent, then just process all three.